pf/internal/schemashim/convert_type.go

```diff
@@ -15,36 +15,43 @@
 package schemashim
 
 import (
+	"context"
 	"fmt"
 
 	pfattr "github.com/hashicorp/terraform-plugin-framework/attr"
 	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
 
 	shim "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim"
 )
 
 func convertType(typ pfattr.Type) (shim.ValueType, error) {
+	// typ.TerraformType is 1:1 to the TF wire format, and is thus closed to outsider
+	// implementation. This allows us to maintain an exhaustive list of possible types and
+	// values.
+	tftype := typ.TerraformType(context.Background())
+	is := func(typ tftypes.Type) bool { return typ.Is(tftype) }
```

Comment:
Is is better than Equal?

Reply:
On the types we are using it for, I believe it is equivalent. Conceptually, `Is` is a shallow comparison, while `Equal` is a deep comparison. Since we are using it on inherently shallow types, there is no difference. I can switch back to `Equal`.